### PR TITLE
feat: permissions

### PR DESCRIPTION
**Pull request by antsgar**
- feat: roles and permissions handling
- tests: tests for permissions service

### CODE REVIEW

**Review by karolsojko — APPROVED** [review]

File: test/permissions.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-undef */
+import { Uuid } from '../packages/snjs/dist/@types/uuid.js';
+import * as Factory from './lib/factory.js';
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+describe('permissions', async function () {
```
Comment (johnny243) [inline test/permissions.test.js:8]: Can we have these tests on `packages/snjs/test`? 🙂 See https://github.com/standardnotes/snjs/commit/bc7f114131241b14022c77484e8ffe3d31528950

Comment (antsgar) [reply test/permissions.test.js:8]: Sure thing @johnny243, thank you for migrating them! I'll merge your branch with this one

**Comment by antsgar** [comment]
Code checks will fail until #355 is merged

**Review by johnny243 — APPROVED** [review]

File: packages/snjs/lib/application.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ import { HttpResponse, SessionListResponse, SignInResponse, User } from './servi
 import { PayloadFormat } from './protocol/payloads';
 import { SNPermissionsService } from './services/permissions_service';
 import { ProtectionEvent } from './services/protection_service';
-import { Permission } from '@standardnotes/auth';
+import { Permission, PermissionName } from '@standardnotes/auth';
```
[the remark this answers is not in the document]
Comment (antsgar) [reply packages/snjs/lib/application.ts]: Done!

**Pull request merged**